packages/plugin-table/src/TableCellNode.ts

```diff
@@ -0,0 +1,170 @@
+import { VNode } from '../../core/src/VNodes/VNode';
+import { TableNode } from './TableNode';
+import { TableRowNode } from './TableRowNode';
+
+// The 'table-cell-attributes' keys of a cell's attributes contains the parsed
+// colspan and rowspan so we can handle them automatically in function of the
+// cell's managed nodes.
+export interface TableCellAttributes extends Record<string, string | Record<string, string>> {
+    'table-cell-attributes'?: Record<string, string> & {
+        parsedColspan: string;
+        parsedRowspan: string;
+    };
+}
+
+export class TableCellNode extends VNode {
+    attributes: TableCellAttributes;
+    header: boolean;
+    // Only the `managerCell` setter should modify the following private keys.
+    _managerCell: TableCellNode;
+    _managedCells = new Set<TableCellNode>();
+
+    constructor(header = false) {
+        super();
+        this.header = header;
+    }
+
+    //--------------------------------------------------------------------------
+    // Lifecycle
+    //--------------------------------------------------------------------------
+
+    /**
+     * Return a new VNode with the same type and attributes as this VNode.
+     *
+     *  @override
+     */
+    clone(): this {
+        const clone = new this.constructor<typeof TableCellNode>(this.header);
+        clone.attributes = { ...this.attributes };
+        return clone;
+    }
+
+    //--------------------------------------------------------------------------
+    // Getters
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    get name(): string {
+        let coordinatesRepr = ' <(' + this.rowIndex + ', ' + this.columnIndex + ')';
+        if (this.colspan > 1 || this.rowspan > 1) {
+            const endRow = this.rowIndex + this.rowspan - 1;
+            const endColumn = this.columnIndex + this.colspan - 1;
+            coordinatesRepr += ':(' + endRow + ', ' + endColumn + ')';
+        }
+        coordinatesRepr += '>';
+        return (
+            super.name +
+            coordinatesRepr +
+            (this.header ? ': header' : '') +
+            (this.isActive ? '' : ' (inactive)')
+        );
+    }
+    /**
+     * Return the cell that manages this cell, if any.
+     */
+    get managerCell(): TableCellNode {
+        return this._managerCell;
+    }
+    /**
+     * Set the given cell as manager of this cell. If null is passed, this cell
+     * will have no manager.
+     */
+    set managerCell(newManager: TableCellNode | null) {
+        const oldManager = this._managerCell;
+        this._managerCell = newManager;
+
+        if (newManager) {
+            // If we just set a new manager for this cell, also add this cell to
+            // the new manager's managed cells.
+            newManager._managedCells.add(this);
+
+            // Copy the manager.
```

Comment:
Well, you're not copying the manager as in "duplicating" him. You're copying the manager *attributes*.

Reply:
No, it is not the same. When you copy/paste on your computer, the original is duplicated. So when you say you "copy the manager", I expect that we end up with a copy *of the manager* somewhere, which is not the case. You never create a *copy* of the manager. It's possible it's mostly my understanding which is biased by computing and "copy" might actually mean what you're saying it means (you know english better than I do), but I still think writing "Copy the manager's attributes and properties." instead of "Copy the manager." would not hurt anything and would improve the understandability for people like me whose english is highly influenced by computing, which is likely to be the case of most people who might stumble upon this comment.

Reply:
Fine, no problem.